Userland/Libraries/LibGfx/BitmapSequence.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2024, Zachary Huang <zack466@gmail.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/RefPtr.h>
+#include <LibGfx/Bitmap.h>
+#include <LibGfx/Size.h>
+#include <LibIPC/Forward.h>
+
+namespace Gfx {
+
+class BitmapSequence {
```

Comment:
Eh. I don't think this class does anything particularly interesting that would make it better than doing

```
namespace Gfx {

using BitmapSequence = Vector<Optional<NonnullRefPtr<Gfx::Bitmap>>>;

}

namespace IPC {

template<>
ErrorOr<void> encode(Encoder&, Gfx::BitmapSequence const&);

template<>
ErrorOr<Gfx::BitmapSequence> decode(Decoder&);

}

```
as the implementation for this header.

Reply:
Hmm, I was originally trying to do something like that, but it wasn't working because of some weird function template specialization rules. Basically, even when I defined `encode` for the type `Vector<Optional<NonnullRefPtr<Gfx::Bitmap>>>` (both as a templated function and as a plain overload), it would end up dispatching to the more generic `Vector<T>` encode function:
```cpp
template<Concepts::Vector T>
ErrorOr<void> encode(Encoder& encoder, T const& vector)
{
    // NOTE: Do not change this encoding without also updating LibC/netdb.cpp.
    TRY(encoder.encode_size(vector.size()));

    for (auto const& value : vector)
        TRY(encoder.encode(value));

    return {};
}
```
instead. I found that when I defined `BitmapSequence` as its own class rather than a type alias, then it would dispatch to the correct function. There might be some C++ wizardry to get around this, but personally I have no idea.

Reply:
Aha, that *is* a good reason! Okay, having a transparent wrapper struct seems fine then. We could define it as 
```
struct BitmapSequence {
    Vector<Optional<NonnullRefPtr<Gfx::Bitmap>>> bitmaps;
};
```
to minimize the boilerplate.